include/misc/mempool.h

```diff
@@ -1,101 +1,13 @@
 /*
- * Copyright (c) 2018 Intel Corporation
+ * Copyright (c) 2019 Intel Corporation
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-
-#ifndef ZEPHYR_INCLUDE_MISC_MEMPOOL_H_
+#ifndef ZEPHYR_INCLUDE_MISC_RING_BUFFER_H_
 #define ZEPHYR_INCLUDE_MISC_MEMPOOL_H_
 
-#include <kernel.h>
-#include <misc/mempool_base.h>
-#include <misc/mutex.h>
-
-struct sys_mem_pool {
-	struct sys_mem_pool_base base;
-	struct sys_mutex mutex;
-};
-
-struct sys_mem_pool_block {
-	struct sys_mem_pool *pool;
-	u32_t level : 4;
-	u32_t block : 28;
-};
+#warning "This header file has moved, include <sys/mempool.h> instead."
```

Comment:
It would be nice to have a Kconfig to disable these warnings. Using the old include is the only way to write programs that are portable with this PR and LTS.

Reply:
yes, those are updated now.

Reply:
Unresolving; there still isn't a Kconfig option for this. Here's what I mean (I prefer option 2).

Option 1: use a bool

```
#ifndef CONFIG_V1_14_INCLUDES
#warning "This header file has moved, include <sys/__assert.h> instead."
#endif
```

That way, applications which need to target multiple Zephyr versions could add `CONFIG_V1_14_STYLE_INCLUDES=y` to their prj.conf.  And actually, if we want to be *really* nice to our users, this option can be `default y` at first, so users can turn it to `n` on their own time. After v1.15, it can be changed to `default n`.

Option 2: use a choice

```
choice
	prompt "Old-style include (v1.14 LTS compatible) support"
	default V1_14_INCLUDES_ALLOW

config V1_14_INCLUDES_ALLOW
	bool "Allow"
	help
	  Allow v1.14-style includes, like <misc/printk.h> instead of
	  <sys/printk.h>

config V1_14_INCLUDES_WARN
	bool "Warn"
	help
	  Emit a warning when v1.14-style includes are used.

config V1_14_INCLUDES_ERROR
	bool "Error"
	help
	  Emit a build error when v1.14-style includes are used.

endchoice
```

This is even better, and provides a way to manage this over multiple versions:

- v.1.15: default is `V1_14_INCLUDES_ALLOW`
- v1.16: default is `V1_14_INCLUDES_WARN`
- v1.17: default is `V1_14_INCLUDES_ERROR`
- v1.18: option is removed and all the shims go away

Reply:
I have seen this (and resolved by mistake), nice proposal, should be discussed before we go ahead and do it this way.